pubtools/_pulp/tasks/set_maintenance/set_maintenance_off.py

```diff
@@ -0,0 +1,44 @@
+import logging
+import re
+
+from .base import SetMaintenance
+from pubtools._pulp.task import PulpTask
+
+
+step = PulpTask.step
+LOG = logging.getLogger("set-maintenance")
+
+
+class SetMaintenanceOff(SetMaintenance):
+    """Unset repositories maintenance mode."""
+
+    @step("Adjust maintenance report")
+    def adjust_maintenance_report(self, report):
+        to_remove = []
+        if self.args.repo_regex:
+            for entry in report.entries:
+                if re.match(self.args.repo_regex, entry.repo_id):
+                    to_remove.append(entry.repo_id)
+        if self.args.repo_ids:
+            existed_repo_ids = [entry.repo_id for entry in report.entries]
+            for repo_id in self.args.repo_ids:
+
+                if repo_id in existed_repo_ids:
+                    to_remove.append(repo_id)
+                else:
+                    LOG.warning("Repository %s is not in maintenance mode", repo_id)
+        LOG.info(
+            "Following repositories will be removed from maintenance mode: \n%s",
+            "\n".join(to_remove),
+        )
+        report = report.remove(to_remove, owner=self.args.owner)
+
+        return report
+
+
+def entry_point(cls=SetMaintenanceOff):
+    cls().main()  # pragma: no cover
```

Comment:
In ClearRepo, the reason why I accepted the class in the entry point was exactly so that I didn't need to omit it from coverage - as I could have at least one test where I passed in my testing subclass of ClearRepo to the entry point.

Here, you've accepted a class as input but then you've also got pragma: no cover.  It looks like in your tests, you did indeed subclass the task classes, so could you also easily cover this entry point by passing it in from at least one test?  Would remove the need for this pragma.

Reply:
I could see you must do this for test coverage, but didn't find out what the real magic is...